src/libopensc/opensc.h

```diff
@@ -562,6 +562,13 @@ struct sc_reader_operations {
 /* Card (or card driver) supports key unwrapping operations */
 #define SC_CARD_CAP_UNWRAP_KEY			0x00001000
 
+/* Card (or card driver) can be initialized using C_InitToken() */
+#define SC_CARD_CAP_ENABLE_INITTOKEN			0x00002000
+/* Card (or card driver) pin unblock options */
+#define SC_CARD_CAP_PIN_UNBLOCK_UNLOGGED_SETPIN		0x00004000
+#define SC_CARD_CAP_PIN_UNBLOCK_SCONTEXT_SETPIN		0x00008000
+#define SC_CARD_CAP_PIN_UNBLOCK_SO_LOGGED_INITPIN	0x00010000
+
```

Comment:
If I see this correctly, then you're not using the new flags in any card driver, so they don't change anything meaningful at the momement, right?

I'm not sure if its a good idea to let the PKCS#11 stuff drip through to the card driver. If the usages of `C_SetPIN` or `C_InitPIN` are ambigious, then there should be an implicit contract between the PKCS#11 application and the PKCS#11 module on how to use it correctly. At the card driver level, however, the uppser layers (PKCS#11) should have sorted out whether a PUK or an SOPIN was passed.